gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -579,6 +614,19 @@ private void announceAllAvailableConnection() {
         }
     }
 
+    private String getAdditionalTimeoutInformation() {
+        final ConnectionResult res = latestConnectionResult.get();
+        if (System.currentTimeMillis() - res.getTime() < cluster.getMaxWaitForConnection() + CONNECTION_SETUP_TIME_DELTA) {
```

Comment:
I'm a bit hesitant to use System.currentTimeMillis() here. I don't like the idea that the returned timeout cause is dependent on when you look it up. If someone ever calls getAdditionalTimeoutInformation() a little while after the timeout has occurred it might give misleading information. Not sure if there is a good alternative though.